idaes/apps/grid_integration/coordinator.py

```diff
@@ -0,0 +1,899 @@
+        Dispatch (SCED), aka "operation".
+
+        Arguments:
+            context: Prescient plugin PluginRegistrationContext from prescient.plugins.plugin_registration
+
+            options: Prescient options from prescient.simulator.config
+
+            plugin_config: Prescient plugin config
+
+        Returns:
+            None
+        """
+
+        context.register_after_ruc_activation_callback(self.activate_DA_bids)
+
+    def _register_finalization_callbacks(
+        self,
+        context,
+        options,
+        plugin_config,
+    ):
+
+        """
+        Register finalization plugins, which run after Prescient simulation
+        finishes.
+
+        Arguments:
+            context: Prescient plugin PluginRegistrationContext from prescient.plugins.plugin_registration
+
+            options: Prescient options from prescient.simulator.config
+
+            plugin_config: Prescient plugin config
+
+        Returns:
+            None
+        """
+
+        context.register_finalization_callback(self.write_plugin_results)
+
+    def initialize_customized_results(self, options, simulator):
+
+        """
+        This method is outdated.
+        """
+
+        simulator.data_manager.extensions["customized_results"] = {}
+        customized_results = simulator.data_manager.extensions["customized_results"]
+
+        customized_results["Generator"] = []
+        customized_results["Date"] = []
+        customized_results["Hour"] = []
+        customized_results["State"] = []
+        customized_results["RUC Schedule"] = []
+        customized_results["SCED Schedule"] = []
+        customized_results["Power Output"] = []
+
+        return
+
+    def _pass_DA_bid_to_prescient(self, options, ruc_instance, bids):
+
+        """
+        This method passes the bids into the RUC model for day-ahead market clearing.
+
+        Arguments:
+            options: Prescient options from prescient.simulator.config.
+
+            ruc_instance: Prescient RUC object
+
+            bids: the bids we want to pass into the day-ahead market. It is a dictionary that has this structure {t: {generator: {power: cost}}}.
+
+        Returns:
+            None
+        """
+
+        gen_name = self.bidder.generator
+
+        # fetch the generator's parameter dictionary from Prescient UC instance
+        gen_dict = ruc_instance.data["elements"]["generator"][gen_name]
+
+        # assemble the bids
+        p_cost = [list(bids[t][gen_name].items()) for t in range(options.ruc_horizon)]
+
+        # update the "p_cost" element in the generator's dict
+        gen_dict["p_cost"] = {
+            "data_type": "time_series",
+            "values": [
+                {
+                    "data_type": "cost_curve",
+                    "cost_curve_type": "piecewise",
+                    "values": p_cost[t],
+                }
+                for t in range(options.ruc_horizon)
+            ],
+        }
+
```

Comment:
```python
        gen_dict["p_min"] = p_min_scalar
```
or
```python
        gen_dict["p_min"] = {'data_type':'time_series', 'values': list_of_pmin}
```
`p_max` can be set similarly.


To fix a generator on
```python
        gen_dict["fixed_commitment"] = 1
```
also fix it to schedule:
```python
        gen_dict["fixed_commitment"] = {'data_type':'time_series', 'values': list_of_0_1_commitments}
        gen_dict["min_up_time"] = 0
        gen_dict["min_down_time"] = 0
```

Could also be set in the real-time updater.